shotover-proxy/src/transforms/cassandra/sink_cluster/routing/mod.rs

```diff
@@ -0,0 +1,2663 @@
+            stream_id: 0,
+            tracing_id: None,
+            warnings: vec![],
+            operation: CassandraOperation::Result(CassandraResult::Prepared(Box::new(
+                BodyResResultPrepared {
+                    id: CBytesShort::new(vec![
+                        11, 241, 38, 11, 140, 72, 217, 34, 214, 128, 175, 241, 151, 73, 197, 227,
+                    ]),
+                    result_metadata_id: None,
+                    metadata: prepared_metadata.clone(),
+                    result_metadata: RowsMetadata {
+                        flags: RowsMetadataFlags::NO_METADATA,
+                        columns_count: 0,
+                        paging_state: None,
+                        new_metadata_id: None,
+                        global_table_spec: None,
+                        col_specs: vec![],
+                    },
+                },
+            ))),
+        }));
+
+        let query_parameters = QueryParams {
+            consistency: One,
+            with_names: false,
+            values: Some(SimpleValues(vec![Value::Some(vec![0, 0, 1, 164])])),
+            page_size: None,
+            paging_state: None,
+            serial_consistency: None,
+            timestamp: None,
+            keyspace: None,
+            now_in_seconds: None,
+        };
+
+        let execute_body = &mut BodyReqExecuteOwned {
+            id: CBytesShort::new(vec![
+                11, 241, 38, 11, 140, 72, 217, 34, 214, 128, 175, 241, 151, 73, 197, 227,
+            ]),
+            result_metadata_id: None,
+            query_parameters: query_parameters.clone(),
+        };
+
+        // let execute = &mut Message::from_frame(Frame::Cassandra(CassandraFrame {
+        //     version: Version::V4,
+        //     stream_id: 0,
+        //     tracing_id: None,
+        //     warnings: vec![],
+        //     operation: CassandraOperation::Execute(Box::new(BodyReqExecuteOwned {
+        //         id: CBytesShort::new(vec![
+        //             11, 241, 38, 11, 140, 72, 217, 34, 214, 128, 175, 241, 151, 73, 197, 227,
+        //         ]),
+        //         result_metadata_id: None,
+        //         query_parameters: query_parameters.clone(),
+        //     })),
+        // }));
+
+        router.add_prepared_result(prepared_result);
+
+        let routing_key = routing_key::calculate(
+            &prepared_metadata.pk_indexes,
+            query_parameters.values.as_ref().unwrap(),
+            Version::V4,
+        )
+        .unwrap();
+
+        assert_eq!(routing_key, vec![0, 0, 0, 4, 0, 0, 1, 164]);
+
+        let nodes: Vec<SocketAddr> = router
+            .execute_message_query_plan(execute_body)
+            .map(|node| node.address)
+            .collect();
+
+        println!("{:?}", nodes);
+
+        let expected_node = "172.16.1.5:9044".parse().unwrap();
+        assert_eq!(nodes.len(), 1);
+        assert_eq!(nodes[0], expected_node);
+    }
+
+    fn prepare_nodes() -> Vec<CassandraNode> {
```

Comment:
For the purposes of reviewing it would help if this was in a separate file so I can minimize the file, otherwise its hard to jump around.